tests/test_dependency_injection.py

```diff
@@ -21,13 +21,31 @@ def havoc_task() -> HavocCheckerTaskMessage:
 
 @pytest.mark.asyncio
 async def test_basic(checker: Enochecker, havoc_task: HavocCheckerTaskMessage):
-    @checker.register_dependency
+    @checker.register_dependency("x")
+    @checker.register_dependency("y")
     def inject_string() -> str:
         return "123"
 
     @checker.havoc(0)
-    async def havoc(param: str):
-        assert param == "123"
+    async def havoc(x_1: str, x_2: str, y: str):
+        assert x_1 == "123" and x_2 == "123" and y == "123"
+
+    await checker._call_havoc(havoc_task)
+
+
+@pytest.mark.asyncio
+async def test_unnamed(checker: Enochecker, havoc_task: HavocCheckerTaskMessage):
+    @checker.register_dependency()
+    def inject_string() -> str:
+        return "123"
+
+    @checker.register_dependency("special")
+    def inject_special_string() -> str:
+        return "456"
+
+    @checker.havoc(0)
+    async def havoc(a: str, b: str, special: str):
```

Comment:
just to clarify, this means we can mix unnamed dependency injections with named dependency injections with the same type and when injecting, if no name matches one of the named dependencies, we use the unnamed dependency injection?

Reply:
yep, the fallback when the name does not match is to look for a 'generic' injector (empty name)

Reply:
hmm, the syntax is a bit different so this is actually still not backwards compatible

Reply:
Tested [9dcc1c7](https://github.com/enowars/enochecker3/pull/27/commits/9dcc1c7692d1c5ca8c47d1ff10efb4fc7378e86f) against enowars5-stldoctor